lib/yggdrasil/player.ex

```diff
@@ -12,9 +13,8 @@ defmodule Yggdrasil.Player do
     Supervisor.add_player user_id, [owner_pid, push_msg]
   end
 
-  def run_cmd(user_id, cmd = %Message{type: :command}) do
-    player_pid = Registry.get_player user_id
-    GenServer.cast(player_pid, {:run_cmd, cmd})
+  def run_cmd(user_id, cmd) do
+    Command.execute cmd, user_id
```

Comment:
Ah yeah wasn't paying attention to position got caught up in the pipeline. 

That seems reasonable to me. You build up this context and the room executes it. 

I like the idea of this expanded structure you just roughed out. I wasn't opposed to just strongly to a vanilla room struct but felt limited. I think what you just proposed would be a great start. 

Then actions like spawn_creature would append the action but the room could do the spawn so unit tests are reduced to just checking the context after a command executes. 

So the player process could execute command then pass the context to something like 

ctx = Command.execute cmd, ctx
Room.run ctx

Or something.